app/controllers/api/v1/application_controller.rb

```diff
@@ -48,14 +48,15 @@ def check_token_and_set_application
       private
 
       def authenticate
-        authenticate_or_request_with_http_token do |token, _options|
+        api_user = authenticate_or_request_with_http_token do |token, _options|
           ApiUser.find_by(token: token)
         end
+
+        Current.api_user = api_user
       end
 
       def current_api_user
```

Comment:
This method is not called in all actions. This means that in many cases `Current.api_user` is not set, and therefore when an audit is created `api_user` is `nil`.